src/dialogs/quick-open/ha-quick-open-dialog.ts

```diff
@@ -0,0 +1,216 @@
+import "@polymer/paper-input/paper-input";
+import "@material/mwc-list/mwc-list-item";
+import "@material/mwc-list/mwc-list";
+import {
+  css,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+} from "lit-element";
+import { fireEvent } from "../../common/dom/fire_event";
+import "../../components/ha-dialog";
+import { haStyleDialog } from "../../resources/styles";
+import { HomeAssistant } from "../../types";
+import { PolymerChangedEvent } from "../../polymer-types";
+import { fuzzySequentialMatch } from "../../common/string/sequence_matching";
+import { componentsWithService } from "../../common/config/components_with_service";
+import { domainIcon } from "../../common/entity/domain_icon";
+import { computeDomain } from "../../common/entity/compute_domain";
+import { domainToName } from "../../data/integration";
+
+export interface QuickOpenDialogParams {
+  entityFilter?: string;
+  commandMode?: boolean;
+}
+
+interface QuickOpenItem {
+  text: string;
+  domain: string;
+  icon?: string;
+  onClick(ev: Event): void;
+}
+
+@customElement("ha-quick-open-dialog")
+export class QuickOpenDialog extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @internalProperty() private _items: QuickOpenItem[] = [];
+
+  @internalProperty() private _entityFilter = "";
+
+  @internalProperty() private _opened = false;
+
+  @internalProperty() private _commandMode = false;
+
+  public async showDialog(params: QuickOpenDialogParams) {
+    this._commandMode = params.commandMode || false;
+    this._opened = true;
+  }
+
+  public closeDialog() {
+    this._opened = false;
+    this._entityFilter = "";
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  private _entityFilterChanged(ev: PolymerChangedEvent<string>) {
+    this._entityFilter = ev.detail.value;
+    if (this._entityFilter.startsWith(">")) {
+      this._commandMode = true;
+      this._entityFilter = this._entityFilter.substring(1);
+    } else {
+      this._commandMode = false;
+    }
+
+    let items: QuickOpenItem[];
+    if (this._commandMode) {
+      items = this._generateCommandItems(this._entityFilter);
+    } else {
+      items = this._generateEntityItems(this.hass, this._entityFilter);
+    }
+    this._items = this._filterItems(items, this._entityFilter);
```

Comment:
Move this to render

Reply:
Filter the items first, then pass them to the render. Now you first process every item and then we are going it throw half of that work away.

Reply:
Actually, a better option for perf would be to generate both item lists _once_ after launching the dialog. Then we can just filter on those cached lists without regenerating the lists each time.

Reply:
Done:

1. Removed the calculation of entity list entirely as the new `renderEntityList` makes it so we don't need to first convert to a common interface
2. Calculate _and store_ the command list only once -- when the dialog launches -- so we don't recalculate after every keystroke
3. Made two different filter methods, one for entities and one for commands
4. Moved filtering logic into `renderCommandList` and `renderEntityList`